pipeline/authn/authenticator_bearer_token.go

```diff
@@ -24,15 +25,29 @@ type AuthenticatorBearerTokenFilter struct {
 }
 
 type AuthenticatorBearerTokenConfiguration struct {
-	CheckSessionURL     string                      `json:"check_session_url"`
-	BearerTokenLocation *helper.BearerTokenLocation `json:"token_from"`
-	PreserveQuery       bool                        `json:"preserve_query"`
-	PreservePath        bool                        `json:"preserve_path"`
-	PreserveHost        bool                        `json:"preserve_host"`
-	ExtraFrom           string                      `json:"extra_from"`
-	SubjectFrom         string                      `json:"subject_from"`
-	SetHeaders          map[string]string           `json:"additional_headers"`
-	ForceMethod         string                      `json:"force_method"`
+	CheckSessionURL       string                      `json:"check_session_url"`
+	BearerTokenLocation   *helper.BearerTokenLocation `json:"token_from"`
+	PreserveQuery         bool                        `json:"preserve_query"`
+	PreservePath          bool                        `json:"preserve_path"`
+	PreserveHost          bool                        `json:"preserve_host"`
+	ExtraFrom             string                      `json:"extra_from"`
+	SubjectFrom           string                      `json:"subject_from"`
+	ForwardHTTPHeaders    []string                    `json:"forward_http_headers"`
+	SetHeaders            map[string]string           `json:"additional_headers"`
+	ForceMethod           string                      `json:"force_method"`
+	ForwardHTTPHeadersMap map[string]string           `json:"-"`
+}
+
+func (a *AuthenticatorBearerTokenConfiguration) ToAuthenticatorForwardConfig() *AuthenticatorForwardConfig {
```

Comment:
A better approach would be to define an interface which is consumed by `forwardRequestToSessionStore`

```
type AuthenticatorForwardConfigProvider interface {
  GetCheckSessionURL() string
  // ...
}

// Type Guard
var _ AuthenticatorForwardConfigProvider = (*AuthenticatorBearerTokenConfiguration)(nil)
```

or alternatively use an embedded struct:

```
type authenticatorForwardConfig struct {
 	CheckSessionURL    string `json:"..."`
}

type AuthenticatorBearerTokenConfiguration struct {
  authenticatorForwardConfig

  // ...
}

// ...
ody, err := forwardRequestToSessionStore(r, cf. authenticatorForwardConfig)
```

That will save some copy & pasting and potential sources of errors (e.g. forgetting to copy a field) and is in general a cleaner approach to Go code :)